pkg/ccl/streamingccl/streamingest/stream_ingestion_processor_test.go

```diff
@@ -171,6 +203,42 @@ func TestStreamIngestionProcessor(t *testing.T) {
 		require.Nil(t, row)
 		testutils.IsError(meta.Err, "this client always returns an error")
 	})
+
+	t.Run("stream ingestion processor shuts down gracefully on losing client connection", func(t *testing.T) {
+		events := []streamingccl.Event{streamingccl.MakeGenerationEvent()}
+		pa := streamingccl.PartitionAddress("partition")
+		mockClient := &mockStreamClient{
+			partitionEvents: map[streamingccl.PartitionAddress][]streamingccl.Event{pa: events},
+		}
+
+		startTime := hlc.Timestamp{WallTime: timeutil.Now().UnixNano()}
+		partitionAddresses := []streamingccl.PartitionAddress{"partition"}
+
+		interceptCh := make(chan struct{})
+		defer close(interceptCh)
+		interceptGeneration := markGenerationEventAsReceived(interceptCh)
+		sip, _, err := getStreamIngestionProcessor(ctx, t, registry, kvDB, "randomgen://test/",
+			partitionAddresses, startTime, []streamclient.InterceptFn{interceptGeneration}, mockClient)
+		require.NoError(t, err)
+
+		var wg sync.WaitGroup
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			sip.Run(ctx)
+		}()
+
+		<-interceptCh
+
+		// Send a cutover signal to shut down the processor
+		sip.cutoverCh <- struct{}{}
+
+		// The processor should have been moved to draining state with a nil error
```

Comment:
We need to move the `wg.Wait() above this so we are guaranteed that `sip.Run()` has returned.

can we use `out` returned by `getStreamIngestionProcessor` to check:
```
if !out.ProducerClosed() {
		t.Fatalf("output RowReceiver not closed")
}

for {
			row := out.NextNoMeta(t)
			if row == nil {
				break
			}
      // We don't expect any rows so...
      t.Fatal(...)
}
```